amethyst_assets/src/loader.rs

```diff
@@ -6,103 +6,50 @@ use std::path::PathBuf;
 use std::sync::Arc;
 
 use fnv::FnvHashMap;
-use futures::{Future, Poll};
+use futures::{Async, Future, IntoFuture, Poll};
+use futures::sync::oneshot::{Receiver, Sender, channel};
 use rayon::ThreadPool;
 use specs::{Component, DenseVecStorage};
 
 use asset::AssetSpec;
 
 use store::AnyStore;
-use {Allocator, Asset, BoxedErr, Context, Directory, Format, AssetError, LoadError, Store};
+use {Allocator, Asset, AssetFuture, BoxedErr, Context, Directory, Format, AssetError, LoadError, Store};
 
 /// Represents a future value of an asset. This future may be
 /// added to the ECS world, where the responsible system can poll it and merge
 /// it into the persistent storage once it is `Ready`.
-pub struct AssetFuture<A, E> {
-    inner: Option<Arc<AssetFutureCell<A, E>>>,
-}
+pub struct SpawnedFuture<A, E>(Receiver<Result<A, E>>);
```

Comment:
This is pretty useful for user-code in order to implement e.g. `create_context`.